specification/devcenter/DevCenter/Environments/routes.tsp

```diff
@@ -122,10 +157,127 @@ interface Environments {
     }
   >;
 
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Missing key"
+  @doc("Gets Outputs from the environment.")
+  @route("/projects/{projectName}/users/{userId}/environments/{environmentName}/outputs")
+  @get
+  @added(APIVersions.v2024_02_01)
+  getOutputs is Azure.Core.Foundations.Operation<
+    {
+      @doc("The DevCenter Project upon which to execute operations.")
+      @maxLength(63)
+      @minLength(3)
+      @pattern("^[a-zA-Z0-9][a-zA-Z0-9-_.]{2,62}$")
+      @path
+      projectName: string;
```

Comment:
Should the project name param in the patchEnvironment operation also have all of these same constraints on it?

Reply:
Regarding this comment. I think we should review all operations and ensure that all path parameters gets the max, min, and pattern added.

Reply:
I noticed afterwards that this stuff is added as "warnings" in [swagger LintDiff](https://github.com/Azure/azure-rest-api-specs/pull/28531/checks?check_run_id=23766920538) but I didn't check this for 2023-04-01 because the pipeline was passing, but I think we should try to fix them

Reply:
Added these constraints to all projectName, poolName, environmentName, catalogName, and slightly different but for userId as well. 

Driele do you know about above and also if devBoxName follows the same constraints? It's not a control plane resource so I'm not sure if it does.

Reply:
Can you also add for OperationId? I believe it's always a guid, so it should be of pattern `"^[a-zA-Z0-9]{8}-([a-zA-Z0-9]{4}-){3}[a-zA-Z0-9]{12}$"` and min and max length 36 (those were wrong in the previous swagger, it had same constraints as projectName)

Reply:
Took a closer look, those are parameter there are still missing constraints: [actionName](https://github.com/Azure/azure-rest-api-specs/blob/e1932ba7a7ad8b3a44ec4bc72f0cb86557dd16c3/specification/devcenter/DevCenter/Environments/models.tsp#L260C1-L263C16), [poolName](https://github.com/Azure/azure-rest-api-specs/blob/e1932ba7a7ad8b3a44ec4bc72f0cb86557dd16c3/specification/devcenter/DevCenter/DevBox/models.tsp#L282C1-L285C16), [scheduleName ](https://github.com/Azure/azure-rest-api-specs/blob/e1932ba7a7ad8b3a44ec4bc72f0cb86557dd16c3/specification/devcenter/DevCenter/DevBox/models.tsp#L405C1-L408C16)

Reply:
everything that has a `@key` decorator in the models we should add the constraints.

Reply:
done